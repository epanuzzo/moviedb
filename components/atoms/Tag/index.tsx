import React from "react";

type TagProps = {
  children: string;
};

/**
 * A functional component that renders a tag with specific styling.
 *
 * @component
 * @param {TagProps} props - The properties passed to the component.
 * @param {React.ReactNode} props.children - The content to be displayed inside the tag.
 * @returns {JSX.Element} A styled span element containing the children.
 */
const Tag: React.FC<TagProps> = ({ children }) => {
  return (
    <span className="inline-block bg-gray-800 rounded-full px-3 py-1 text-sm mr-2 my-2">
      {children}
    </span>
  );
};

export default Tag;
