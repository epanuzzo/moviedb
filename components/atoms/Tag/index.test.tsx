import React from "react";
import renderer from "react-test-renderer";
import Tag from "./index";

describe("Rating", () => {
  it("renders component", () => {
    const tree = renderer.create(<Tag>Test</Tag>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
