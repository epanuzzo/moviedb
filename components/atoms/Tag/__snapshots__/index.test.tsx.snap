// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders component 1`] = `
<span
  className="inline-block bg-gray-800 rounded-full px-3 py-1 text-sm mr-2 my-2"
>
  Test
</span>
`;
